src/main/java/org/stellar/sdk/Asset.java

```diff
@@ -17,67 +17,68 @@
    */
   public static Asset create(String canonicalForm) {
     if (canonicalForm.equals("native")) {
-      return new AssetTypeNative();
+      return create(canonicalForm, null, null);
     }
     String [] parts = canonicalForm.split(":");
     if (parts.length != 2) {
       throw new IllegalArgumentException("invalid asset "+ canonicalForm);
     }
-    return Asset.createNonNativeAsset(parts[0], parts[1]);
+    return create(null, parts[0], parts[1]);
   }
 
   /**
-   * Creates Asset for Alpha4/Alpha5/Native
+   * Creates Asset for Alpha4/Alpha12/Native
    *
-   * @param type the type of asset can be 'native', 'alpha4', 'alpha12'
-   * @param code the asset code that conforms to type or null
-   * @param issuer the asset issuer the conforms to type or null
-   * @return
+   * @param type the type of asset. 'native' will generate its respective asset sub-class,
+   *             if null or any other value will attempt to derive the asset sub-class from code and issuer.
+   * @param code the asset code or null
+   * @param issuer the asset issuer or null
+   * @return Asset
    */
   public static Asset create(String type, String code, String issuer) {
     return create(type, code, issuer, null);
   }
 
   /**
-   * Creates Asset for Alpha4/Alpha5/Native/LiquidityPool
+   * Creates Asset for Alpha4/Alpha12/Native/LiquidityPool
    *
-   * @param type the type of asset can be 'native', 'alpha4', 'alpha12' or 'liquidity_pool_shares'
-   * @param code the asset code that conforms to type or null
-   * @param issuer the asset issuer the conforms to type or null
-   * @param liquidityPoolID provided only if type is 'liquidity_pool_shares'
+   * @param type the type of asset. 'native' and 'liquidity_pool_shares' will generate their respective asset sub-classes
+   *             null or any other value will attempt to derive the asset sub-class from code and issuer.
+   * @param code the asset code or null
+   * @param issuer the asset issuer or null
+   * @param liquidityPoolID required only if type is 'liquidity_pool_shares'
    * @return Asset
    */
   public static Asset create(String type, String code, String issuer, String liquidityPoolID) {
+    if (type == null) {
```

Comment:
what's the rationale for changing the behavior of `type` to allow null values?

previously: type the type of asset can be 'native', 'alpha4', 'alpha12' or 'liquidity_pool_shares'

Reply:
https://github.com/stellar/go/blob/d550cca61c47b1b0bd3f9cc4fe1350d6e69b0f68/services/horizon/internal/assets/main.go#L17-L21

horizon uses credit_alphanum4 and credit_alphanum12 instead of alpha4 and alpha12

Reply:
the alpha4, alpha12 type values were/are ignored, rather the asset code is introspected to determine the Asset class type, only type values of 'native' and 'liquidity_pool_shares' have effect on derived Asset class, otherwise, clients don't need to pass a specific 'alpha4, alpha12' type, for backwards compat, it's fine, if it's one of those values or null will result in the same introspection of asset code to determine Asset class.

Reply:
trying to stay with existing static factory method pattern, but one other creational pattern to consider, but thought it might be too much breaking change, is to remove the `create` methods, and replace with builder pattern, such as:
`Asset.newBuilder().withCode().withIssuer().withLiquidityPoolId().create()`

if you don't include any `withXyz()` then by default the builder creates a native.